chatGPT_API/Controller/StoryViewController.swift

```diff
@@ -0,0 +1,172 @@
+        
+        model.fetch(with: queryManager.continueQuery(story: currentTitle, option: currentOption))
+    }
+    
+    @IBAction func choiceButtonPressed(_ sender: Any) {
+        self.performSegue(withIdentifier: "historyPopUp", sender: self)
+    }
+    
+}
+
+//MARK: - Story Delegate
+//delegate to update the story once response came to DomainModel
+extension StoryViewController: StoryManagerDelegate{
+    func didUpdateStory(story: String) {
+        let modeledStory = storyBrain.modelAStory(input: model.content)
+        if modeledStory?.title == "" || (modeledStory?.title.count)! < 15{
+            print("Title is empty. It's wrong and I'm trying not to change it")
+            model.fetch(with: queryManager.continueQuery(story: currentTitle, option: currentOption))
+        }else{
+            self.option1.setTitle(modeledStory?.choice1, for: .normal)
+            self.option1.sizeToFit()
+            self.option2.setTitle(modeledStory?.choice2, for: .normal)
+            self.option2.sizeToFit()
+            self.option3.setTitle(modeledStory?.choice3, for: .normal)
+            self.option3.sizeToFit()
+            storyAnimate(storyLabel: self.storyTitle, storyTitle: modeledStory!.title)
+        }
+    }
+}
+    
+//MARK: - Animate
+//Letter by letter animated text
+extension UILabel {
+    func animate(newText: String, characterDelay: TimeInterval, completion: (() -> Void)? = nil) {
+        DispatchQueue.main.async() {
+            self.text = ""
+            for (index, character) in newText.enumerated() {
+                DispatchQueue.main.asyncAfter(deadline: .now() + characterDelay * Double(index)) {
+                    self.text?.append(character)
+                    if index == newText.count - 1 { // check if this is the last character to be animated
+                    completion?() // call the completion handler if it's not nil
+                }
+                }
+            }
+        }
+    }
+}
+//MARK: - Animations
+extension StoryViewController{
+    func hideLabel(){
+        UIView.animate(withDuration: 0.5, animations: {
+            self.storyTitle.alpha = 0
+        })
+    }
+    
+    func showLabel(){
+        UIView.animate(withDuration: 0.5, animations:{
+            self.storyTitle.alpha = 100
+        })
+    }
+    func hideButton() {
+        
+        UIView.animate(withDuration: 0.5, delay: 0.5, animations: {
+            self.option1.alpha = 0.0
+            self.option2.alpha = 0.0
+            self.option3.alpha = 0.0
+        })
+    }
+    
+    // When the function is triggered again, animate the alpha back to 1.0
+    func showButton() {
+        UIView.animate(withDuration: 0.5, animations: {
+            self.option1.alpha = 1.0
+        })
+        UIView.animate(withDuration: 0.5, delay: 0.35, animations: {
+            self.option2.alpha = 1.0
+        })
+        UIView.animate(withDuration: 0.5, delay: 0.7, animations: {
+            self.option3.alpha = 1.0
+        })
```

Comment:
Given how often alpha is being changed, it can be safer and provide more information to use an extension on CGFloat:
```
extension CGFloat {
    static let alphaVisible: CGFloat = 1.0
    static let alphaHidden: CGFloat = 0.0
}

self.option3.alpha = .alphaHidden
self.option3.alpha = .alphaShown

```